src/components/grid/grid.ts

```diff
@@ -176,6 +176,15 @@ export class Grid extends Component<Props, SpreadsheetChildEnv> {
         target: this.env.model.getters.getSelectedZones(),
       });
     },
+    ESCAPE: () => {
+      if (this.env.model.getters.hasOpenedPopover()) {
+        this.closeOpenedPopover();
```

Comment:
is the getter really necessary to reach your goal ? we have several calls to this function inside the component and none of them takes the time to check if the popover is opened. We should either check everytwhere or never do it, don't you think ?

Reply:
> is the getter really necessary to reach your goal ? we have several calls to this function inside the component and none of them takes the time to check if the popover is opened. We should either check everytwhere or never do it, don't you think ?

@rrahir If we want to keep this order of closing/cleaning something, I would say yes. The idea is when there's a popover, hitting ESC will close it (and will not clean the clipboard selection). If there's no popover but a context menu open, hitting ESC closes the context menu.

Reply:
my point is that you could call both functions 100% of the time without checking that it's necessary beforehand, just like we do in other places in the same file. I you want to have a "check" step before closing the opened popovr then it should be used widely accrosse the `Grid` component, not just here ;-)

Reply:
@rrahir 
Yes, it's not necessary to check beforehand if we want to close something. 
Here `if...else...` is used to make sure that if there is a clipboard selection and a open popover together it will only close the popover, and not clean the clipboard selection (GSheets behaves this way). I need to put something in the `if` statement so that's why the getter is needed. 
If hitting ESC once to close the open popover and clean clipboard selection at the same time is what we want, then the getter is not needed.

Reply:
Ok my bad, I must be partly blind.
Anyway since you have made that getter, please use it to wrap other calls to `this.closeOpenedPopover();` it's not a big optimisations but we don't need a full dispatch when it's not required.

Could you add a note at the top of the piece of code ?
/** TODO:  Clean once we introduce proper focus on sub components. Grid should not have to handle all this logic */